### PR TITLE
Accept a ServiceProvider instance when creating a new AssemblyLoadContext.

### PR DESCRIPTION
**Pull request by troydai**
Addressing #1777

### CODE REVIEW

File: src/Microsoft.Framework.Runtime/Loader/AssemblyLoadContextFactory.cs [hunk 1]
```diff
@@ -8,19 +8,17 @@ namespace Microsoft.Framework.Runtime.Loader
...
 
         public AssemblyLoadContextFactory(IServiceProvider serviceProvider)
         {
             var accessor = serviceProvider.GetService(typeof(IAssemblyLoadContextAccessor)) as IAssemblyLoadContextAccessor;
-            _serviceProvider = serviceProvider;
             _defaultContext = accessor.Default;
         }
-
-        public IAssemblyLoadContext Create()
+
```
Comment (moozzyk) [inline src/Microsoft.Framework.Runtime/Loader/AssemblyLoadContextFactory.cs]: Use this: https://visualstudiogallery.msdn.microsoft.com/bcb3ef8e-a3a4-401e-9e63-fdf00b45268f

File: src/Microsoft.Framework.DesignTimeHost/DesignTimeAssemblyLoadContextFactory.cs [hunk 2]
```diff
@@ -14,36 +14,33 @@ internal class DesignTimeAssemblyLoadContextFactory : IAssemblyLoadContextFactor
...
             _project = project;
             _appEnv = appEnv;
-            _hostServices = hostServices;
             _cache = cache;
             _cacheContextAccessor = cacheContextAccessor;
             _namedDependencyProvider = namedDependencyProvider;
         }
-
-        public IAssemblyLoadContext Create()
+
```
Comment (BrennanConroy) [inline src/Microsoft.Framework.DesignTimeHost/DesignTimeAssemblyLoadContextFactory.cs]: Space

**Comment by troydai** [comment]
@davidfowl approved. I'll merge it soon.

**Pull request merged**